Rtl Skin

seyyedhasan6172 opened the pull request:
#7883 
hi 
I change styles of the elastic theme directly to set it RTL.as you say in issue #7883 check the codes and say your opinion about how I change my code to be useful for your needs
I change default styles to rtl styles and add text-align:right and direction:rtl to htm,body tags in style .css

seyyedhasan6172 committed:
change default styles to rtl styles and add text-align:right and direction:rtl to htm,body tags in style .css

alecpl commented:
Sorry but this way I don't see what has changed in styles. You should not modify styles.css, but the less files.

seyyedhasan6172 committed:
2021-03-05
I change .less files for Rtl, for changing styles I change styles.less and styles that imported in styles.less
,global.less
,layout.less
,common.less
,buttons.less
,jqueryui.less
,dialogs.less
,menu.less
,messages.less
,lists.less
,forms.less
,editor.less
 and so some other that imported in up .less files

alecpl commented:
Thanks. From the changes I see that this probably should be done in the same way we implemented dark mode (see dark.less). I.e. create rtl.less file and put all needed less changes under `html.rtl` class (or better `html[dir=rtl]`) and make sure to set the class/attribute with some code in the templates engine (depending on the selected language?).

seyyedhasan6172 committed:
2021-03-18
create rtl.less file.
 I have created rtl.less like the dark.less .I have tested in other skin and its works. I change the html.rtl to html.dark-mode and I have import rtl.less instead of dark.less.
When I was clicked on the dark-mode button the direction changed to RTL and هt worked properly.
so. I want to change the direction according to the language. I see some functions in editor.js for detect language, if I detect the language in ui.js I can change the direction. I think I can implement like dark-mode scenario.
if you have an idea that what can I do please say it to me.
Thanks.

seyyedhasan6172 commented:
Hello
 I have created rtl.less like the dark.less .I have tested in other skin and its works. I change the html.rtl to html.dark-mode and I have import rtl.less instead of dark.less.
When I was clicked on the dark-mode button the direction changed to RTL and هt worked properly.
so. I want to change the direction according to the language. I see some functions in editor.js for detect language, if I detect the language in ui.js I can change the direction. I think I can implement like dark-mode scenario.
if you have an idea that what can I do please say it to me.
Thanks.

seyyedhasan6172 committed:
Edited

seyyedhasan6172 commented:
hi 
Have you checked the codes?
Were they in your opinion or not?
I put the codes inside the dark.less and the direction changed by pressing the theme button.

alecpl commented:
Could you remove changes that are not needed anymore?

seyyedhasan6172 commented:
hi 
what do you mean?
Currently the important file is rtl.less. Which changes should I delete?

alecpl commented:
See "Files changed" tab here. To review changes I have to see only these relevant.

seyyedhasan6172 committed:
Delete layout.less

seyyedhasan6172 committed:
Revert "Delete layout.less"

This reverts commit deaf3333601758f5ee42edce4d2aeebef26973d7.

seyyedhasan6172 committed:
Revert "change default styles to rtl styles and add text-align:right and direction:rtl to htm,body tags in style .css"

This reverts commit b96dc4107c750cc79be7ff45c0e43dd588fa3c82.

seyyedhasan6172 committed:
Revert "2021-03-05"

This reverts commit a0ddc537d19649691abae73e18f512638e4f19f4.

alecpl commented:
Ok, so now before anyone can test this we need to add at the end of styles.less:
```
& when (@rtl-mode-enabled = true) {
    @import "rtl";
}
```
and define the variable in variables.less, where `@dark-mode-enabled` is.

Next steps:
1. Set `rtl` class on the html element depending on the UI language of the current user (`rcmail_output_html::_write()`).
2. Do we need to set dir="rtl" anywhere?
3. Investigate if we need to do anything with TinyMCE. I.e. switch to RTL mode automatically. What if the edited content (e.g. a mail response) is in English, what if it's in Arabic.
4. Investigate what other things need to be fixed/improved.

seyyedhasan6172 committed:
define @rtl-mode-enabled in variables.less
add condition for check @rtl-mode-enabled at the end of styles.less
add a function and call that for change direction

seyyedhasan6172 committed:
Merge pull request #2 from seyyedhasan6172/rtl_branch

rtl condition

seyyedhasan6172 commented:
I define @rtl-mode-enabled in variables.less,
add a condition for check @rtl-mode-enabled at the end of styles.less,
add a function and call that for change direction
if you change the language to Persian, you can see that direction changed.

alecpl commented:
This looks better and better. I didn't yet tested this, but do you have any feedback on points 2-4 from my last comment?

Neustradamus commented:
@seyyedhasan6172: Have you progressed on your PR?

@alecpl has requested you comment on some points.

101t commented on skins/elastic/ui.js:
```diff
@@ -4244,6 +4248,14 @@ function rcube_elastic_ui()
     };
+
+    function changeDirection() {
+        lang = rcmail.env.locale;
```
```var lang = rcmail.env.locale;```
set var because js is restricted

danielmayost commented:
Is there an update on this issue? (Sometimes a small addition/change to the code can simultaneously apply to another 2 billion people in the world). 

Thank you.
